### PR TITLE
Journal and xpub integration via Selenium, click on 2 buttons

### PR DESCRIPTION
**Pull request by giorgiosironi**
Minimal scope so far:
```
spectrum/test_xpub.py [2018-10-16 16:03:16,039][INFO][spectrum.input][] Loading: https://end2end--journal.elifesciences.org
[2018-10-16 16:03:17,857][INFO][spectrum.input][journal] Found submit button
[2018-10-16 16:03:22,047][INFO][spectrum.input][journal] Clicked submit button
[2018-10-16 16:03:22,074][INFO][spectrum.input][elife-xpub] Found login button
[2018-10-16 16:03:23,281][INFO][spectrum.input][elife-xpub] Clicked login button
.[2018-10-16 16:03:23,281][INFO][spectrum.generator][] Deleting Selenium driver <selenium.webdriver.remote.webdriver.WebDriver (session="b9ca26ba-570e-461c-b060-30578a67e7
17")>
```

Depends on https://github.com/elifesciences/elife-xpub-deployment/pull/25 at least.

### CODE REVIEW

File: spectrum/input.py [hunk 1]
```diff
@@ -202,13 +202,56 @@ def __init__(self, host, cdn_host):
...
+
+    def _log(self, message, *args, **kwargs):
+        LOGGER.info(message, extra={'app':'journal'}, *args, **kwargs)
+
+    def submit(self):
+        LOGGER.info("Loading: %s", self._host)
+        self._driver.get(self._host)
+        selenium_title_smoke_test('eLife', self._driver)
+
+        submit_link = self._driver.find_element_by_link_text('SUBMIT MY RESEARCH')
```
Comment (thewilkybarkid) [inline spectrum/input.py]: Can use the ID now.

File: spectrum/input.py [hunk 2]
```diff
@@ -202,13 +202,56 @@ def __init__(self, host, cdn_host):
...
+        self._log("Found submit button")
```
Comment (thewilkybarkid) [inline spectrum/input.py]: Sounds like a form. 'Submit my research'?

File: spectrum/input.py [hunk 3]
```diff
@@ -202,13 +202,56 @@ def __init__(self, host, cdn_host):
...
+        submit_link.click()
+        self._log("Clicked submit button")
```
Comment (thewilkybarkid) [inline spectrum/input.py]:
```suggestion
        self._log("Clicked 'Submit my research' button")
```

(Any way to do a multi-line suggestion?)

Comment (giorgiosironi) [reply spectrum/input.py]: May work:

Will expand log with the new DOM id

**Comment by giorgiosironi** [comment]
After switching to the id, logs now look like:
```
[2018-10-17 14:32:51,790][INFO][spectrum.input][journal] Found #submitResearchButton `SUBMIT MY RESEARCH`
[2018-10-17 14:32:54,267][INFO][spectrum.input][journal] Clicked #submitResearchButton
[2018-10-17 14:32:54,324][INFO][spectrum.input][elife-xpub] Found login button button[data-test-id="login"] `LOGIN WITH ORCID`
[2018-10-17 14:32:55,567][INFO][spectrum.input][elife-xpub] Clicked login button button[data-test-id="login"]
```

**Comment by giorgiosironi** [comment]
Will have to wait on some elife-xpub operational fixes before being merged, for stability.

**Pull request merged**